gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/EhrResourceExtractorUtil.java

```diff
@@ -65,6 +65,34 @@ public static boolean hasEhrFolder(RCMRMT030101UK04Component component) {
         return component.getEhrFolder() != null;
     }
 
+    public static RCMRMT030101UK04EhrComposition extractEhrCompositionForNarrativeStatement(RCMRMT030101UK04EhrExtract ehrExtract,
+        II resourceId) {
+        return ehrExtract.getComponent()
+            .stream()
+            .filter(EhrResourceExtractorUtil::hasEhrFolder)
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .filter(EhrResourceExtractorUtil::hasEhrComposition)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .filter(ehrComposition -> filterForMatchingEhrCompositionNarrativeStatement(ehrComposition, resourceId))
+            .findFirst()
+            .get();
+    }
+
+    public static List<RCMRMT030101UK04EhrComposition> extractValidDocumentReferenceEhrCompositions(RCMRMT030101UK04EhrExtract ehrExtract) {
```

Comment:
This function needs to incorporate what Elliot has worked on in also extracting valid NarrativeStatements that are part of nested CompoundStatements